lib/utils/import/evernote/index.ts

```diff
@@ -37,84 +28,28 @@ class EvernoteImporter extends EventEmitter {
       return;
     }
 
-    const saxStream = createStream(true, false);
-    const parser = new DOMParser();
     const coreImporter = new CoreImporter({
       noteBucket: this.noteBucket,
       tagBucket: this.tagBucket,
     });
-    let currentNote = {}; // The current note we are parsing
-    let importedNoteCount = 0;
-
-    saxStream.on('error', function () {
-      this.emit('status', 'error', 'Error processing Evernote data.');
-    });
-
-    saxStream.on('opentag', (node) => {
-      // The <note> tag signifies that we should parse another note
-      if (node.name === 'note') {
-        currentNote = { tags: [] };
-      }
-    });
-
-    saxStream.on('cdata', (text) => {
-      // Note content in evernote exports lives in CDATA
-      const htmlDoc = parser.parseFromString(text, 'text/html');
 
-      // We're only interested in 'note' doctypes, like 'en-note'
-      if (!endsWith(get(htmlDoc, 'doctype.name', ''), 'note')) {
+    const addNotesToApp = (notes) => {
+      if (!notes.length) {
+        this.emit(
+          'status',
+          'error',
+          'Error processing Evernote data or it did not contain any notes.'
+        );
         return;
       }
-
-      const strippedText = enmlToMarkdown(htmlDoc.body.innerHTML);
-      if (strippedText !== '') {
-        currentNote.content += '\n' + strippedText;
-      }
-    });
-
-    saxStream.on('text', (text) => {
-      if (!text) {
-        return;
-      }
-
-      const tagName = saxStream._parser.tagName;
-      switch (tagName) {
-        case 'title':
-          // Evernote titles appear to be plain text only, we can take it as-is
-          currentNote.content = text;
-          break;
-        // We need to convert the date to a Unix timestamp
-        case 'created':
-          currentNote.creationDate = this.getConvertedDate(text);
-          break;
-        case 'updated':
-          currentNote.modificationDate = this.getConvertedDate(text);
-          break;
-        case 'tag':
-          currentNote.tags.push(text);
-      }
-    });
-
-    saxStream.on('closetag', (node) => {
-      // Add the currentNote to the array
-      if (node === 'note') {
-        coreImporter.importNote(currentNote, this.options);
-        importedNoteCount++;
-        this.emit('status', 'progress', importedNoteCount);
-      }
-    });
-
-    saxStream.on('end', () => {
-      if (importedNoteCount === 0) {
-        this.emit('status', 'error', 'No notes were found to import.');
-        return;
-      }
-
-      this.emit('status', 'complete', importedNoteCount);
-    });
-
-    // Read the file via stream
-    fs.createReadStream(file.path).pipe(saxStream);
+      notes.forEach((note) => {
+        coreImporter.importNote(note, this.options);
+      });
+      this.emit('status', 'complete', notes.length);
+    };
+
+    window.electron?.once('notesImported', addNotesToApp);
+    window.electron?.send('importNotes', file.path);
```

Comment:
unlike other parts of the app where integrations may or may not be present there is no way to make the Evernote importer work without Electron, right? 

if this is true then I think we need to hide it when not in an Electron environment and abort if `window.electron` isn't defined. that means we shouldn't need to use the optional chaining for safety and can rely on `window.electron.once()`.

that will probably leave a type error here but if we let that error stay then it will be a reminder for us to either fix it deeper in our types and how we structure the code or to remember that if something fails it could be because we're assuming `window.electron` is there (vs. hiding that fact in a graceful fail for a process that directly depends on it).

Reply:
I can add a check to the top of the file to return if it is not Electron. This code should never be executed outside of Electron: https://github.com/Automattic/simplenote-electron/blob/develop/lib/dialogs/import/source-selector.tsx#L17